lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -79,6 +79,10 @@ data TransactionLayer t = TransactionLayer
         -- This estimate will err on the side of permitting more inputs,
         -- resulting in a transaction which may be too large.
 
+    , validateSelection
+      :: CoinSelection
+      -> Either ErrMkStdTx ()
```

Comment:
Can we really return all the error cases from `ErrMkStdTx`? It may be convenient to re-use the same error type but we are probably conveying a misleading information. It'd be better to have another error type `ErrInvalidTx` or something like this.

Reply:
Added `ErrValidateSelection` to comply with naming convention introduced by `ErrMkStdTx`